R/data.table.R

```diff
@@ -2554,8 +2554,17 @@ setnames <- function(x,old,new) {
   invisible(x)
 }
 
-setcolorder <- function(x, neworder)
+setcolorder <- function(x, neworder=NULL)
 {
+  if (is.null(neworder)) {
+    keys = key(x)
+	if (is.null(indices(x))) {
+		indcs = c()
+	} else {
+		indcs = unique(setdiff(do.call(c, indices(x, vectors=T)), keys))
+	}
+    neworder = c(keys, indcs, setdiff(names(x), c(keys, indcs)))
+  }
   if (any(duplicated(neworder))) stop("neworder contains duplicates")
   # if (!is.data.table(x)) stop("x is not a data.table")
   if (length(neworder) != length(x)) {
```

Comment:
@mattdowle `setcolorder(x)` in current master will error here. so no old behavior broken